src/main/java/uk/gov/hmcts/darts/courthouse/controller/CourthousesController.java

```diff
@@ -61,7 +62,7 @@ public ResponseEntity<AdminCourthouse> adminCourthousesCourthouseIdGet(
 
             return new ResponseEntity<>(adminCourthouse, HttpStatus.OK);
         } catch (EntityNotFoundException | JpaObjectRetrievalFailureException | IllegalStateException e) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+            throw new DartsApiException(CourthouseApiError.COURTHOUSE_NOT_FOUND);
```

Comment:
Consider adding a unit test to cover these exceptions, and adding a log.error message before each one with the 'e' variable passed in so the stack trace can be seen, otherwise this information will be lost. We saw similar issue in model court testing last week (elsewhere in code) and was a nightmare to work out what had actually caused the issue. https://tools.hmcts.net/jira/browse/DMP-2433 has been raised already to address but if we're changing this code now, makes sense to add it.

Reply:
I've checked and amended the exception handling here and added a couple of unit tests.